### DISCUSSION TITLE
nextjs logging on docker

### DISCUSSION THREAD

**Discussion by fabioportieri** [vercel/next.js#57502] [Help]
### Summary

Hello, i'm deploying two nextjs app on docker, but i noticed i have no logs whatsoever in the backend, my console.logs do not get printed

my dockerfile is pretty much the one in 
https://medium.com/@elifront/best-next-js-docker-compose-hot-reload-production-ready-docker-setup-28a9125ba1dc

is there a configuration to do, in order to be able to see logs in a dockerized application?
the documentation here: 
https://nextjs.org/docs/pages/building-your-application/deploying/production-checklist#logging
mentions the usage of pino, i'll try that too but i'm not very optimistic

### Example

my dockerfile is a readaptation of this:

https://github.com/vercel/next.js/blob/canary/examples/with-docker-compose/next-app/prod.Dockerfile

**Comment by alzalabany** [comment]
same, i'm considering droping next just because debuger and console.log never works!

**Comment by fabioportieri** [comment]
i opened an issue with a minimal test case here:
https://github.com/vercel/next.js/issues/60723

**Comment by jahidem** [comment]
mine showing nothing! complete blank. Can't debug like this.[😭](https://emojipedia.org/loudly-crying-face)
![image](https://github.com/user-attachments/assets/c1a865d6-f1b2-4fb6-9f4c-b3195d846201)

**Comment by YannicHock** [comment]
I found that if the invoking page is opting out of the Full Route Cache [using one of these options](https://nextjs.org/docs/app/building-your-application/caching#opting-out-2), console.log() goes through to the container log. While this will not be realy usefull for production builds, this might help you out whilst debugging until there is a proper logger implementation for next.js
